### ISSUE TITLE
Uploading large assets results in clunky verification 

### ISSUE THREAD

**Issue by Fabric-Digital** [craftcms/cms#5779]
The customer uploads 1000-1500 (sometimes a lot more) photos to an entry asset field.
 
The status bar that first pops up is perfect. It shows the customer how long it will take for the images to upload.

HOWEVER, once the status bar goes away, the asset selector box comes up. Since they technically need to select all, they need to wait for “All” of the thumbnails to load, and they need to keep scrolling to the bottom of the dialogue box in order to select all of the assets

Would love a way for the user to easily upload large amounts of assets at once.

**Comment by brandonkelly** [comment]
I’ll leave this open, as this does reveal a real flaw in the UX, but you should be able to upload + select in one go by dragging the images directly onto the Assets field. (Though I’m not very confident about how efficiently it will perform for _1000+_ assets at once.)

**Comment by Fabric-Digital** [comment]
Yeah we do that.

But it’s the second step that matters, there really needs to be a select all field in the asset preview window so we don’t have to scroll all the way down to the bottom using the scroll bar (which only populates like 50 at a time).

Regards,
Ryan
0212989127

________________________________

**Comment by RitterKnightCreative** [comment]
@Fabric-Digital I can relate.

I'm not sure the exact use case you're describing but if you're, say, a photographer or a publisher who might use Craft for proofing images, or what not, that workflow gets really old, fast. 

Even an option to forgo the thumbnails if there's more than a couple pages worth and display all the assets in a scrolling list with the entirety of them already checked could work. (As an example, I think NextGen for WP does this but it's been awhile since I used it.)

FWIW you could probably write a plugin to handle this too. IMO it'd be ideal if the client uploads a single zip to the assets field, Craft would detect it contains assets inside, and then offer to include all of them in one go. Craft would be also be smart enough to unpack the ZIP, even if your assets field only specified JPEGs, basically using your computer to do the selecting.

**Comment by brandonkelly** [comment]
@Fabric-Digital I meant drag the files right onto the field itself, **not** the asset selection modal. (Don’t click the “Add an asset” button.) That way the files will get auto-added to the field selection.

**Comment by Fabric-Digital** [comment]
> @Fabric-Digital I meant drag the files right onto the field itself, **not** the asset selection modal. (Don’t click the “Add an asset” button.) That way the files will get auto-added to the field selection.

Ahah! I think you may be on to something. Let me test and get back to you.

**Comment by Fabric-Digital** [comment]
> @Fabric-Digital I can relate.
> 
> I'm not sure the exact use case you're describing but if you're, say, a photographer or a publisher who might use Craft for proofing images, or what not, that workflow gets really old, fast.
> 
> Even an option to forgo the thumbnails if there's more than a couple pages worth and display all the assets in a scrolling list with the entirety of them already checked could work. (As an example, I think NextGen for WP does this but it's been awhile since I used it.)
> 
> FWIW you could probably write a plugin to handle this too. IMO it'd be ideal if the client uploads a single zip to the assets field, Craft would detect it contains assets inside, and then offer to include all of them in one go. Craft would be also be smart enough to unpack the ZIP, even if your assets field only specified JPEGs, basically using your computer to do the selecting.

Yeah we are thinking about the plugin route, we just need a solution in the interim. I think @brandonkelly may have given us the interim solution we need.

**Comment by Fabric-Digital** [comment]
@brandonkelly that interim workaround mitigated the second step which sometimes took 5 - 10 mins per entry. Thank you!
